Core/DISET/private/Service.py

```diff
@@ -60,7 +61,16 @@ def __init__( self, serviceData ):
     if serviceData[ 'loadName' ] not in self._validNames:
       self._validNames.append( serviceData[ 'loadName' ] )
     self._cfg = ServiceConfiguration( list( self._validNames ) )
-    if serviceData[ 'standalone' ]:
+    # ElasticDB
+    self.activityMonitoring = gConfig.getValue("/DIRAC/ActivityMonitoring", "false").lower() in ("yes","true") 
+    gLogger.info("The value fo activitMonitoring flag is " + str(self.activityMonitoring))
+    if self.activityMonitoring:
+      self.activityMonitoringReporter = MonitoringReporter(monitoringType="ComponentsActivityMonitoring")
+      self.log = {}
```

Comment:
I am assuming this is a dictionary which contains the entries you want to send to the monitoring ? `log` is not really an appropriate name :)
Moreover, `self.log` is a instance attribute which is modified in multiple threads if I understand correctly -> not thread safe

Reply:
So how does this look if we transfer all the ES monitoring logic to a method named `_activityMonitoring` and inside that we have a dictionary which is only accessible within this method and we call it periodically with `gThreadScheduler.addPeriodicTask(300, _activityMonitoring)` this way ? 
And can you please suggest a name instead of `log` ?

Reply:
For the name, something more describing, I don't know, like `monitoringData` or, `monitoringRecord` or anything like that. 

I am not sure I understand your logic of period task. For example, we want to increase the query counter everytime there is a connection. How would you do that with a periodic task ?

Reply:
So can we update it with the help of this `self._stats[ 'connections' ]` which is accessible globally?
With the help of which we can update both connections and query counter.